src/qt/walletmodel.cpp

```diff
@@ -692,6 +694,34 @@ bool WalletModel::abandonTransaction(uint256 hash) const
     return wallet->AbandonTransaction(hash);
 }
 
+bool WalletModel::transactionSignalsRBF(uint256 hash) const
+{
+    LOCK2(cs_main, wallet->cs_wallet);
+    const CWalletTx *wtx = wallet->GetWalletTx(hash);
+    if (wtx || SignalsOptInRBF(*wtx))
+        return true;
+    return false;
+}
+
+bool WalletModel::bumpFee(uint256 hash, const std::function<bool(const CTransaction& tx, const CAmount& oldFee, const CAmount& newFee)> verificationCallback) const
+{
+    LOCK2(cs_main, wallet->cs_wallet);
+    CAmount nOldFee(0);
+    CAmount nNewFee(0);
+    std::shared_ptr<CWalletTx> wtxRef;
+    std::vector<std::string> vErrors;
+
+    // do a simple feebump with default confirmation target
+    CWallet::BumpFeeResult res = wallet->BumpFee(hash, 0, false, 0, true, nOldFee, nNewFee, wtxRef, vErrors, verificationCallback);
```

Comment:
I'm surprised it's ok to be holding onto the cs_main and cs_wallet locks while waiting for user input from a confirmation dialog, since it would seem to grind the whole bitcoin node to a halt. But maybe there are other precedents for doing this.

If you wanted to avoid it, though, I think you easily could by splitting bumpfee into two functions: one which creates the new transaction, and the other which commits this.